src/js/tools/pen.js

```diff
@@ -66,28 +65,32 @@ define(function (require, exports, module) {
                     return descriptor.playObject(optionsObj);
                 });
 
-        if (this.vectorMode) {
-            var selectVectorMask = descriptor.playObject(vectorMaskLib.activateVectorMaskEditing());
-            return Promise.join(setPromise, selectVectorMask);
-        } else {
-            var deleteFn = function (event) {
-                event.stopPropagation();
+        var deleteFn = function (event) {
+            event.stopPropagation();
+            if (toolStore.getVectorMode()) {
+                this.flux.actions.mask.handleDeleteVectorMask();
+            } else {
+                this.flux.actions.menu.native({ commandID: _CLEAR_PATH });
+            }
+        }.bind(this);
 
-                return PS.performMenuCommand(_CLEAR_PATH)
-                    .catch(function () {
-                        // Silence the errors here
-                    });
-            };
+        var backspacePromise = this.transfer(shortcuts.addShortcut,
+                OS.eventKeyCode.BACKSPACE, {}, deleteFn, "penBackspace", true),
+            deletePromise = this.transfer(shortcuts.addShortcut,
+                OS.eventKeyCode.DELETE, {}, deleteFn, "penDelete", true),
+            disableSuppressionPromise = UI.setSuppressTargetPaths(false),
```

Comment:
It looks like you're calling `setSuppressTargetPaths(false)` twice if you're in mask mode and once if you're not. Do you want to call this once regardless of the mode? Or do you only want to call it once in mask mode?